src/lib.rs

```diff
@@ -43,3 +46,61 @@ pub mod futures {
     pub use crate::rwlock::{Read, UpgradableRead, Upgrade, Write};
     pub use crate::semaphore::{Acquire, AcquireArc};
 }
+
+/// The strategy for polling an `event_listener::EventListener`.
```

Comment:
imo it would make sense if this would be put into `event-listener` instead (maybe behind a feature flag)

Reply:
Maybe? A similar strategy is used in `async-channel`, so putting it in another crate would make sense. On the other hand, it's not really the most idiomatic API, and exposing it to users would give me pause.

Reply:
this is why I'm suggesting a feature flag. could be maybe additionally versioned (because it might be likely that it would need more releases and fine-tuning than the `event-listener` core itself), but for that, a separate crate (idk `event-listener-strategy`) might be appropriate (although it would be probably not a good idea to not develop it in tandem with `event-listener`, so it might make sense to put it into the `event-listener` repo as just yet another crate, and turn the repo into a workspace.

Reply:
I'd be fine with it, but I think this needs more discussion.

@smol-rs/admins Thoughts on splitting the `Strategy` trait into a new crate related to `event-listener`?

Reply:
I've opened smol-rs/event-listener#49 to implement this.

Reply:
@notgull nice! will be useful for https://github.com/smol-rs/async-broadcast/issues/41 as well.

Reply:
@notgull looks like we are missing a publish on the event-listener-strategy, for this to be updated

Reply:
Publishing the strategy crate is blocked on the v3.0.0 release of `event-listener`, which will probably take a while.